src/Specifications/formatDate.js

```diff
@@ -0,0 +1,63 @@
+const formatDate = ( string ) => {
+  let result = [],
+    monthName = { Jan:'01', Feb:'02', Mar:'03', Apr:'04', May:'05', Jun:'06', Jul:'07', Aug:'08', Sep:'09', Oct:'10', Nov:'11', Dec:'12' },
+    filtered,
+    split,
+    year
+
+
+    if ( string.includes('-') ) {
+      return filterString(string)
+    }
+
+    if ( string.includes('/') ) {
+      filtered = filterString(string)
```

Comment:
There's a lot of code repetition in these `if` blocks - every one of them repeats these same two lines:
```js
filtered = filterString( string )
split = splitString( filtered )
```
These lines could be removed from the individual `if` blocks.  The `if` blocks could would then perform the format specific operations necessary to return the correct string.

As well, each of those specific operations could be refactored into meaningfully named functions that clearly communicate what transformation they are applying in order to correct the string.